quinn/src/connection.rs

```diff
@@ -38,12 +38,41 @@ where
 
 impl<S> Connecting<S>
 where
-    S: proto::crypto::Session,
+    S: proto::crypto::Session + 'static,
 {
-    pub(crate) fn new(conn: ConnectionRef<S>, connected: oneshot::Receiver<bool>) -> Self {
-        Self {
+    pub(crate) fn new(
```

Comment:
I'm fully on board with encapsulating the coupling of the driver and `Connecting`. However, this change appears to closely couple `Connecting` and the `ConnectionRef`, and this seems unnecessary to achieve the former goal (we could still make `ConnectionRef::new()` private to the `connection` module, of course).

Reply:
I'm a bit concerned about the growth of the `ConnectionRef::new` arguments list, but it's not untenable just yet. Applied.